tarides/tezos
# Introduce Irmin stats within Tezos

This PR is a draft for exposing the Irmin statistics from the `irmin-pack` library within Tezos through the Prometheus system.
It is a continuation of mirage/irmin#1574.

To be able to test this feature, you need to:
- Have Prometheus running on your machine
  ```
  sudo systemctl start prometheus
  ```
- Ensure that you have copied this text at the end of `/etc/prometheus/prometheus.yml`:
   ```yml
    - job_name: 'netdata-scrape'
      metrics_path: "/api/v1/allmetrics"
      params:
        format: [prometheus]
      static_configs:
        - targets: ['localhost:19999']
    - job_name: 'tezos-exporter'
      metrics_path: "/metrics"
      static_configs:
        - targets: ['localhost:9091']
    ```
- Run a tezos node with `--metrics-addr="localhost:9091"`
